### ISSUE TITLE
RKE2/K3s cluster cannot provision successfully using cluster scoped registry

### ISSUE THREAD

**Issue by catherineluse** [rancher/rancher#39338]
Seen in v2.6-head. To reproduce:

1. Install Rancher normally with the default private registry
2. Go to the global settings and set `system-default-registry` to an invalid global registry. I'm calling it `invalid.io`
3. Provision an RKE2 cluster in Digital Ocean. In the Registries section of the cluster configuration form, enter a valid private registry. I'm using `docker.io`:
<img width="910" alt="Screen Shot 2022-10-07 at 2 09 10 PM" src="https://user-images.githubusercontent.com/20599230/194659287-9a367386-9d30-4dea-9d42-339a6e0b5e5b.png">
4. If you click **Edit as YAML**, it appears as if the registry has been correctly configured at `rkeConfig.machineSelectorConfig[0].config['system-default-registry']`:
<img width="727" alt="Screen Shot 2022-10-07 at 2 33 12 PM" src="https://user-images.githubusercontent.com/20599230/194660645-f36d3f82-1409-4140-9d2e-4eedc4fdf1a0.png">
5. Finish creating the cluster
6. The cluster gets stuck in provisioning mode, with the message "Waiting for viable init node"

<img width="1418" alt="Screen Shot 2022-10-07 at 2 35 27 PM" src="https://user-images.githubusercontent.com/20599230/194661376-55cb07a0-bce8-44ec-b6f8-5ea08646b36c.png">
7. If you delete the cluster from the UI, it doesn't get removed. It just gets stuck waiting for the machine to delete.

The expected behavior is that the cluster gets created successfully because its images are supposed to be pulled from the valid registry, and it is supposed to be able to be deleted even if it gets stuck.

If the above mentioned YAML config is invalid, it's a frontend issue. If it's valid, it's a backend issue.

**Comment by gaktive** [comment]
Transferred this from Dashboard repo since the UI appears to be doing the correct thing here with the YAML.

**Comment by HarrisonWAffel** [comment]
I've looked into this a bit and it seems like this is due to how we have implemented machine provisioning with CAPI.

When provisioning RKE1 clusters, the `rancher-machine` binary is stored locally called directly (either from a jail or from the local `/usr/bin`). 

When provisioning RKE2 clusters, the `rancher-machine` image is pulled from a registry during the execution of a machine-provision job, and the image name is prefixed with the global registry name [here](https://github.com/rancher/rancher/blob/9f1e0435bb45e104ba897c5c329e1f23735843d9/pkg/controllers/provisioningv2/rke2/machineprovision/args.go#L169). So, when an invalid global registry is set, RKE2 will fail to pull the `rancher-machine` image and provisioning will be unable to proceed.

A potential fix for this issue is to incorporate the cluster-level registry when pulling the `rancher-machine` image during the machine-provision job. Although, it can be argued that the current behavior is by design, as it is the rancher server which provisions nodes and not the downstream cluster. The rancher server does not have a concept of a cluster level registry, only the global registry.

**Comment by catherineluse** [comment]
> it can be argued that the current behavior is by design, as it is the rancher server which provisions nodes and not the downstream cluster

Ah, that makes sense. Is there any way a user could tell from the UI if the cluster scoped registry was successfully used during cluster provisioning, or if the global registry was used as a fallback for all images?

**Comment by Josh-Diamond** [comment]
@catherineluse Yes, if the user visits `Cluster Explorer > Pods > All namespaces`, the user will be able to verify where images were pulled

**Comment by catherineluse** [comment]
OK. In that case, this issue isn't blocking any UI work getting done. To me it would seem reasonable to close it as expected behavior since, as Harrison pointed out, node provisioning does seem outside of the cluster scope and would require the global registry to be valid. So it is expected that both cluster registry and global registry would have to be valid.

**Comment by Josh-Diamond** [comment]
Yes, I agree.  Closing out this issue, as it has been discovered this is not a valid or supported use case
